### PR TITLE
Refactor/GitHub#13

### PR DESCRIPTION
**Pull request by monshan**
# Pull Requests for TikTok Organizer Niche Project

## Summary of changes

- Applies the following refactoring suggestions from PR #13 

  1. Creates mock.js file to hold the mock data originally in App
  2. Refactors <code>addPin</code> & <code>removePin</code> to a single <code>togglePin</code> that re-assigns isPinned to the opposite of its current value with the <code>!</code> operator
  3. Refactors ternary conditional rendering in Independent to a method on Independent with a single if conditional and default return

## Notes for reviewer (any known bugs, areas of interest, etc.)

- Home component has since been removed in a previous branch

## Next Steps (Future Iterations)

- Reapply PropTypes with updated prop requirements to reflect architectural changes

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -5,29 +5,12 @@ import AddTikTokForm from '../AddTikTokForm/AddTikTokForm';
...
-    'https://www.tiktok.com/@clintonaverytharp/video/6954587026595351813?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0',
-    'https://www.tiktok.com/@csapunch/video/6954532471551642885?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0',
-    'https://www.tiktok.com/@bananalovesyoutoo/video/6945128725625294086?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0',
-    'https://www.tiktok.com/@forkthatsgoood/video/6953763533813271813?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=00',
-    'https://www.tiktok.com/@abz248/video/6951099476664716550?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0',
-    'https://www.tiktok.com/@carejeffcounty/video/6943239331855355142?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0',
-    'https://www.tiktok.com/@goldenretrieverlife/video/6954103546321161478?sender_device=pc&sender_web_id=6925894707823576582&is_from_webapp=v1&is_copy_url=0'
-  ])
+  const [initTikToks, setInitTikToks] = useState(mock);
```
Comment (piknikki) [inline src/components/App/App.js:13]: nice! so clean!!

File: src/components/App/App.js [hunk 2]
```diff
@@ -65,13 +48,9 @@ const App = () => {
...
-    setDisplayHome([...sortIfPinned(fetchedTTS)]);
-  }
-
-  const removePin = (id) => {
-    fetchedTTS[fetchedTTS.findIndex((tiktok) => tiktok.data_video_id === id)].isPinned = false;
+  const togglePin = (id) => {
+    const position = fetchedTTS.findIndex(tiktok => tiktok.data_video_id === id);
+    fetchedTTS[position].isPinned = !fetchedTTS[position].isPinned;
     setDisplayHome([...sortIfPinned(fetchedTTS)]);
   }
```
Comment (piknikki) [inline src/components/App/App.js:56]: That looks great. So DRY!

**Review by piknikki — COMMENTED** [review]
Great work! I won't merge, since I think you want Connor to do a review, but it looks ready.
🐢 🔥

**Comment by monshan** [comment]
@piknikki amazing! and no worries I'm going to go ahead and merge and if he makes further comments I can create another branch for it

File: src/components/Independent/Independent.js [hunk 3]
```diff
@@ -25,12 +25,31 @@ const Independent = ({
...
+  togglePin
   }) => {
 
-  const togglePin = () => {
-    isPinned ? removePin(data_video_id) : addPin(data_video_id)
+  const determineErrorRender = () => {
+    if (status_msg) {
+      return (
+        <CardContent>
+          <p>Not able to load this tiktok, please check your path url</p>
```
Comment (ConnorAndersonLarson) [inline src/components/Independent/Independent.js:35]: Love the clear and concise error message but I think it is getting too technical for the average tiktok user. Would they know what the path url is/should be and know what to do to fix it? 

Might be more non-tech friendly to have them return to the home page and try again or just try a different video.

**Pull request merged**